eng/common/scripts/Package-Properties.ps1

```diff
@@ -76,81 +76,76 @@ function Get-PkgProperties
 {
     Param
     (
-        [Parameter(Mandatory = $true)]
```

Comment:
This might be a good candidate for [Parameter Sets in PowerShell](https://devblogs.microsoft.com/powershell/powershell-v2-parametersets/). The general idea is that if you have a set of arguments that form a set of mutually exclusive options, then parameter sets might help.

Reply:
Am not sure that handles what am trying to do here. It looks like that is for allowing one to pass the arguments without specifying the name and letting PowerShell figure out the right argument. For my case it looks like what is needed is [Dynamic Parameters](https://www.powershellmagazine.com/2014/05/29/dynamic-parameters-in-powershell/) but the last time I tried to use that, @weshaggard said it is unnecessarily complicates the code so we switched back to the simple if statement.

Reply:
In the code on line 86 you throw an error that says that "You must specify either PacakgeName or ArtifactName". This is the classic scenario for parameter sets where you have, here is a concise example:

```powershell
# Do-Stuff.ps1
param(
	[Parameter(ParameterSetName="p1", Mandatory=$true)]
	$FirstOnly,
	[Parameter(ParameterSetName="p2", Mandatory=$true)]
	$SecondOnly,
	[Parameter(Mandatory=$true)]
	$Both
)

Write-Host $PsCmdlet.ParameterSetName
```

With the invocation:

```.\Do-Stuff.ps1 -FirstOnly x -Both y```

... the output would be ```p1```, and the output for this invocation would be ```p2```:

```.\Do-Stuff.ps1 -SecondOnly x -Both y```

The idea is that you use PowerShell to figure out which kind of invocation you are doing, and then you can just have a big conditional at the top.

Reply:
While you could do parameter sets I don't think it is very interesting when the set only has one parameter as you the conditions would do in code are pretty much the same in that case. It is much more interesting when the parameter sets are larger and mutually exclusive.